chunky/src/java/se/llbit/chunky/renderer/FrameSampler.java

```diff
@@ -145,15 +189,17 @@ protected final void postprocess(double[] pixel, double exposure, Postprocess po
    * @throws IOException
    */
   public static FrameSampler read(DataInput inStream) throws IOException {
-    byte samplerType = inStream.readByte();
+    Type samplerType = Type.from(inStream.readByte());
 
     switch (samplerType) {
-      case SAMPLER_TYPE_UNIFORM:
+      case UNIFORM:
         return new UniformFrameSampler(inStream);
-      case SAMPLER_TYPE_ADAPTIVE:
+      case ADAPTIVE:
         throw new NotImplementedException();
       default:
-        throw new IOException("Invalid frame sampler type: " + (int)samplerType);
+        // if the tag is not a valid type, Type.from() will throw an exception, otherwise we're
+        // guaranteed a valid Type, which means this default statement should never be reached.
+        throw new RuntimeException("Reached unreachable statement");
```

Comment:
I guess this can be removed then.

Reply:
I didn't have this at first but without it Java complains that all code paths don't return a value :confused: